pkg/ddl/create_table.go

```diff
@@ -264,15 +271,17 @@ func onCreateTables(jobCtx *jobContext, job *model.Job) (int64, error) {
 	if err != nil {
 		return ver, errors.Trace(err)
 	}
-
-	job.State = model.JobStateDone
-	job.SchemaState = model.StatePublic
-	job.BinlogInfo.SetTableInfos(ver, tableInfos)
 	for i := range tableInfos {
 		createTableEvent := notifier.NewCreateTableEvent(tableInfos[i])
-		asyncNotifyEvent(jobCtx, createTableEvent, job)
+		err = asyncNotifyEvent(jobCtx, createTableEvent, job, w.sess)
```

Comment:
one job can only insert one event, because there's PK constraint in the storage layer. Maybe change NewCreateTableEvent to NewCreateTablesEvent.

Reply:
Do you mean we should add a new schema change event whose type is `ActionCreateTables`

Reply:
We should create a new event type and maybe delete the event of `ActionCreateTable` to avoid caller only uses one of them

Reply:
Oh, i see, you mean use `NewCreateTablesEvent` replace the  `NewCreateTableEvent`, and the create single table event can also be covered

Reply:
updated, ptal again [2755f94](https://github.com/pingcap/tidb/pull/56425/commits/2755f949efa06ab20ed3ba6630a79ac7cb26e0a3)

Reply:
/cc @Rustin170506 @time-and-fate  @winoros please also take a look at this commit